app/mailers/school_mailer.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class SchoolMailer < ApplicationMailer
+  NOMINATION_EMAIL_TEMPLATE = "a7cc4d19-c0cb-4187-a71b-1b1ea029924f"
+
+  def nomination_email(params)
+    template_mail(
+      NOMINATION_EMAIL_TEMPLATE,
+      to: params[:recipient],
+      reference: params[:reference],
+      rails_mailer: mailer_name,
+      rails_mail_template: action_name,
+      personalisation: {
+        school_name: params[:school_name],
+        nomination_link: params[:nomination_url],
+      },
+    )
+  end
```

Comment:
Hmm, I don't like the blanket `params` in a method definition. 

I'm happy for it to be a hash (so we don't need to remember an arbitrary order of several arguments), but I would still like the parameter names to be specified.

Also, I do wonder, are some of the computations we do in the rake task going to be needed each time we send a nomination email? Like making a new reference or deciding which school email we send the message to? If so, maybe we want to pass a `school` to this method, and do those calculations here.

Reply:
I don't wan't to add any of our logic in the mailer. These are just for sending the email and I wan't them to be free of the record keeping or deciding who to send to or when to send and when not to send.

Reply:
I don't see anything wrong with the hash tbh. Passing in a hash is so ubiquitous when ones wants flexibility with the arguments e.g. model.new / model.create. What is it you don't like about it?

Reply:
I want the method signature to look like
```
def nomination_email(recipient:, reference:, school_name:, nomination_url:)
```
so I don't need to read the method body to understand what fields I should include in my hash.

Reply:
Just curious what your cutoff point would be with the number of arguments. I love keyword arguments too but I'd probably move to a hash if there were at least 4 arguments. Happy to change it

Reply:
My cut off point is more to do with "how many different things does the method need to know to function?". 

If a method needs to know about 10 things, and they are all important, distinct, and can't be grouped, then I prefer having 10 named arguments over a single `params`. Having single `params` saves me some time and space when writing the method signature, but it makes it a lot harder to read, and therefore to use effectively.

So really, I try to limit the number of things a method needs to know about. That's where I was going with my suggestion to put some logic in the mailer - so we could pass a `school` rather than `school_name` and `recipient`, which are quite closely tied together in our data. 

For mailers a lot of things that we end up passing is personalisation - if we had like 10 of those, maybe we should think about computing it in the mailer (and unit testing it there), or about computing it somewhere else and passing the complete personalisation object to the mailer. Not sure we need it right now, just a thought :)